src/super_gradients/training/datasets/detection_datasets/yolo_format_detection.py

```diff
@@ -198,18 +201,26 @@ def _load_annotation(self, sample_id: int) -> dict:
         return annotation
 
     @staticmethod
-    def _parse_yolo_label_file(label_file_path: str) -> np.ndarray:
+    def _parse_yolo_label_file(label_file_path: str, ignore_invalid_labels: bool = True) -> np.ndarray:
         """Parse a single label file in yolo format.
 
         #TODO: Add support for additional fields (with ConcatenatedTensorFormat)
+        :param label_file_path:         Path to the label file in yolo format.
+        :param ignore_invalid_labels:   Whether to ignore labels that fail to be parsed. If True ignores and logs a warning, otherwise raise an error.
 
         :return: np.ndarray of shape (n_labels, 5) in yolo format (LABEL_NORMALIZED_CXCYWH)
         """
         with open(label_file_path, "r") as f:
-            labels_txt = f.read()
+            lines = f.readlines()
 
         labels_yolo_format = []
-        for line in labels_txt.split("\n"):
-            label_id, cx, cw, w, h = line.split(" ")
-            labels_yolo_format.append([int(label_id), float(cx), float(cw), float(w), float(h)])
-        return np.array(labels_yolo_format)
+        for line in filter(lambda x: x != "\n", lines):
+            try:
+                label_id, cx, cw, w, h = line.split(" ")
+                labels_yolo_format.append([int(label_id), float(cx), float(cw), float(w), float(h)])
+            except Exception as e:
+                if ignore_invalid_labels:
+                    logger.warning(f"Line `{line}` of file {label_file_path} will be ignored because not in LABEL_NORMALIZED_CXCYWH format: {e}")
+                else:
+                    raise e
```

Comment:
I added an option to ignore when the label could not be parsed properly (and log a warning).
I can definitely imagine this happening from time to time.
But maybe we want to handle it differently and raise an explicit error instead of just dropping the label.
Any thought?